src/pytest_servers/factory.py

```diff
@@ -266,11 +266,9 @@ def gcs(
             **kwargs,
         )
         path.fs.mkdir(bucket_name, enable_versioning=version_aware, exist_ok=False)
-
         # UPath adds a trailing slash here, due to which
         # gcsfs.isdir() returns False.
         # pylint: disable=protected-access,assigning-non-slot
-        original = path._accessor._format_path  # noqa: SLF001
-
-        path._accessor._format_path = lambda path: original(path).rstrip("/")  # type: ignore[method-assign] # noqa: SLF001
+        _ = str(path)  # force Path to compute `Path._str`
+        path._str = path._str.rstrip("/")  # remove trailing slash
```

Comment:
`upath` now returns a new instance of accessor each time you access those. Also, `UPath._accessor` usage is now deprecated, although it is what is used internally.

I haven't found a correct place to fix this without extending, as 
`Upath.path` are properties that just ask `Pathlib.path` which in turn just calls `__str__` (which is adding a trailing slash to it).

Ideally, `gcfs` should support a path with a trailing slash, but `Upath` should not add a trailing slash when it's not provided either. I'll try to create an issue on both repositories.

Reply:
Thanks @skshetry, can you add a note here pointing to those issues?

Reply:
Okay, looking at s3fs, it does not handle `s3fs.info("bucket/")` or `s3fs.info("bucket")` either. It raises `FileNotFoundError` in both cases.

However, it does magically work with `_exists()` due to a separate implementation.
As we already do `.mkdir()`, the bucket is created. So I don't think we should try to support something that is not supported by `gcsfs` at all. 

The users of this library need not worry about whether the bucket is created or not. I'm skipping this ugly workaround for now.